report/sla.go

```diff
@@ -126,6 +126,26 @@ func SLAText(probers []probe.Prober) string {
 	return text
 }
 
+// SLALogSection return the Log format string to stat
+func SLALogSection(r *probe.Result) string {
+	text := `name="%s"; endpoint="%s"; up="%s"; down="%s"; sla="%.2f%%"; total="%d(%s)"; latest_time="%s"; latest_status="%s"; message="%s"`
+	return fmt.Sprintf(text, r.Name, r.Endpoint,
+		DurationStr(r.Stat.UpTime), DurationStr(r.Stat.DownTime), r.SLAPercent(),
+		r.Stat.Total, SLAStatusText(r.Stat, Log),
+		time.Now().UTC().Format(r.TimeFormat),
+		r.Status.String(), r.Message)
+}
+
+// SLALog return a full stat report with Log format
+func SLALog(probers []probe.Prober) string {
+	var text string
+	n := len(probers)
+	for i, p := range probers {
+		text += fmt.Sprintf("SLA-Report-%d-%d %s\n", i+1, n, SLALogSection(p.Result()))
```

Comment:
propose to use `bytes.Buffer` to improve performance in case there are many probers.

Reply:
Almost all of the SLA functions use the string + operation,  I will address this issue for all functions in another dedicated PR.